distributed/client.py

```diff
@@ -130,6 +131,46 @@
 TOPIC_PREFIX_FORWARDED_LOG_RECORD = "forwarded-log-record"
 
 
+def _clean_excepthook(func):
+    @wraps(func)
+    def wrapper(exc_type, exc, tb):
+        tb = truncate_traceback(tb)
+        return func(exc_type, exc.with_traceback(tb), tb)
+
+    return wrapper
+
+
+def _clean_ipython_traceback(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        exc_type, exc, tb = sys.exc_info()
+        tb = truncate_traceback(tb)
+        kwargs["exc_tuple"] = exc_type, exc.with_traceback(tb), tb
+        value = func(*args, **kwargs)
+        return value
+
+    return wrapper
+
+
+original_excepthook = sys.excepthook
+sys.excepthook = _clean_excepthook(sys.excepthook)
+
+
+# if ipython is available, also customize showtraceback
+try:
+    from IPython.core.interactiveshell import InteractiveShell
+except ImportError:
+    pass
+else:
+    InteractiveShell.showtraceback = _clean_ipython_traceback(
+        InteractiveShell.showtraceback
+    )
+
+
+def _restore_exception_hook():
+    sys.excepthook = original_excepthook
+
```

Comment:
Is it needed to restore the original `InteractiveShell.showtraceback` as well then?

Reply:
Good point, added.